### PR TITLE
SymfonyContainer::getInstance() null in webservice Image/Product delete

### LINKED ISSUE #32897: Deleting product image doesn't work via API/webservice
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

As title says, image isn't deleted.
Error says
```
Fatal error: Uncaught Error: Call to a member function get() on null in /var/www/html/classes/Image.php:640
Stack trace:
#0 /var/www/html/classes/Image.php(210): ImageCore->deleteImage()
#1 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(822): ImageCore->delete()
#2 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(625): WebserviceSpecificManagementImagesCore->manageDeclinatedImagesCRUD(true, '/var/www/html/i...', Array, '/var/www/html/i...')
#3 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(650): WebserviceSpecificManagementImagesCore->manageEntityDeclinatedImages('/var/www/html/i...', Array)
#4 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(656): WebserviceSpecificManagementImagesCore->manageDeclinatedImages('/var/www/html/i...')
#5 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(315): WebserviceSpecificManagementImagesCore->manageProductImages()
#6 /var/www/html/classes/webservice/WebserviceSpecificManagementImages.php(184): WebserviceSpecificManagementImagesCore->manageImages()
#7 /var/www/html/classes/webservice/WebserviceRequest.php(619): WebserviceSpecificManagementImagesCore->manage()
#8 /var/www/html/webservice/dispatcher.php(94): WebserviceRequestCore->fetch('8A7LBRETHEZ9PUS...', 'DELETE', 'images/products...', Array, false, NULL)
#9 {main}
  thrown in /var/www/html/classes/Image.php on line 640
```

https://github.com/PrestaShop/PrestaShop/assets/23308427/8dacd8c2-d869-4f7f-aabc-dcdf3c10916a

### Expected behavior

Image is deleted.

### Steps to reproduce

1. Generate API key
2. Obtain product image ID 
3. Call /api/images/products/PRODUCT_ID/IMAGE_ID with DELETE method.

### PrestaShop version(s) where the bug happened

dev/master

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by alexll**
When deleting Image or Product with images using the webservice SymfonyContainer::getInstance() is null.

Commit solves issue #32897

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | SymfonyContainer::getInstance() is null in webservice Image delete and webservice Product delete. When deleting Image or Product with images using the webservice SymfonyContainer::getInstance() is null. Commit solves issue #32897
| Type?             | bug fix
| Category?         | WS
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Delete product image or product with images using the webservice
| Fixed ticket?     | Fixes #32897
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

File: classes/Image.php [hunk 1]
```diff
@@ -638,7 +638,12 @@ public function deleteImage($forceDelete = false)
         // Delete auto-generated images
         $image_types = ImageType::getImagesTypes();
         $sfContainer = SymfonyContainer::getInstance();
-        $isMultipleImageFormatFeatureEnabled = $sfContainer->get(FeatureFlagStateCheckerInterface::class)->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT);
+
+        if ($sfContainer instanceof ContainerInterface) {
+	        $isMultipleImageFormatFeatureEnabled = $sfContainer->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT);
+        }else {
+	        $isMultipleImageFormatFeatureEnabled = false;
```
Comment (boherm) [inline classes/Image.php:645]: You must still use `FeatureFlagStateCheckerInterface` to retrieve the right state of the feature flag.
In fact, if you use the repository, you only retrieve the state in database, but now in `develop` branch, you can set feature flag states with env, dotenv or databse. So you must use the state checker to be sure to retrieve the right state.

File: classes/Image.php [hunk 2]
```diff
@@ -941,17 +946,21 @@ public function getPathForCreation()
...
         // Old 2x thumbnail, if present
         $filesToDelete[] = $this->image_dir . $this->getExistingImgPath() . '-' . $imageType['name'] . '2x.' . $imageFormat;
 
-        // Watermarked thumbnail, if present
-        if ($configuration->get('WATERMARK_HASH')) {
-            $filesToDelete[] = $this->image_dir . $this->getExistingImgPath() . '-' . $imageType['name'] . '-' . $configuration->get('WATERMARK_HASH') . '.' . $imageFormat;
+        $sfContainer = SymfonyContainer::getInstance();
+
+        if ($sfContainer instanceof ContainerInterface) {
+	        $configuration = $sfContainer->get('prestashop.adapter.legacy.configuration');
```
Comment (boherm) [inline classes/Image.php:958]: We are avoid string to get service from the container.
For `prestashop.adapter.legacy.configuration`, you can use instead:
```suggestion
	        $configuration = $sfContainer->get(Configuration::class);
```
Don't forget to add `use PrestaShop\PrestaShop\Adapter\Configuration` at the top of this file.

**Review by boherm — CHANGES_REQUESTED** [review]
Hi @alexll! Thanks for you contribution!

I'm not really sure about your PR, I think we must find first of all why the `SymfonyContainer` is null before doing such conditions everywhere.
What do you think about it? :)

**Comment by alexll** [comment]
Thank's for the review @boherm!

SymfonyContainer is null due to that the call initiated form webservice and kernel is not loaded in this legacy system.

You can find more details in issue #20740 and a possible solution in the pull https://github.com/PrestaShop/PrestaShop/pull/26205 but seems it was dismissed in https://github.com/PrestaShop/PrestaShop/pull/26205#pullrequestreview-830170348

Maybe someone with more expertise in symfony services can contribute a better solution.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Hi, we cant just NOT delete the images if the container is not available, we must find another solution. :-)

**Review by matks — COMMENTED** [review]
For the record https://github.com/PrestaShop/PrestaShop/pull/24403 introduced a boot of the kernel for every Webservice request (I was not very happy about that 😄 ) and an optimization was added in https://github.com/PrestaShop/PrestaShop/pull/27766

But in this PR there is something that puzzles me 🤔 : is the container `every time` or are there usecases where it is null and usecases where it is not null?

**Comment by alexll** [comment]
Container is only null when called from a webservice request. 

When functions are called using the back office it works as expected.

As you mentioned the kernel is not initialised when using webservices.

A similar fix like https://github.com/PrestaShop/PrestaShop/pull/27766 should be applied for Images .

**Comment by Hlavtox** [comment]
The issue will be fixed by https://github.com/PrestaShop/PrestaShop/pull/33407, if somebody approves the PR and it gets merged

**Pull request closed without merging**